grpc/src/main/java/com/linecorp/armeria/internal/client/grpc/ArmeriaClientCall.java

```diff
@@ -195,16 +204,9 @@ public void start(Listener<O> responseListener, Metadata metadata) {
                 close(status, new Metadata());
             } else {
                 ctx.setResponseTimeout(TimeoutMode.SET_FROM_NOW, Duration.ofNanos(remainingNanos));
-                ctx.whenResponseCancelling().thenAccept(cause -> {
```

Comment:
question: where is this logic moved?

Reply:
The `TimeoutException` will be propragated to `HttpStreamDeframerHandler.processOnError(Throwable)` via publisher-subscriber chains. And it is converted to `DEADLINE_EXCEEDED` staus by GrpcStatus.fromThrowable().
https://github.com/line/armeria/pull/2981/files#diff-c2e9a6ba40a5a8ccd62caf83de8f01a6b4ac1b305cc47b2bb07a3eb1c5f3b2c4R122-R124

Reply:
Thanks for the explanation. 🙏